src/test/java/com/exasol/rls/administration/scripts/AssignRolesToUserIT.java

```diff
@@ -60,42 +61,53 @@ protected Connection getConnection() throws NoDriverFoundException, SQLException
+                Arguments.of(List.of("role_1", "role_2"), 3), //
+                Arguments.of(List.of("role_1", "role_4"), 9), //
+                Arguments.of(List.of("role_1", "role_2", "role_3", "role_4"), 15), //
+                Arguments.of(List.of("role_2", "role_3", "role_53", "role_63"),
+                        BitField64.ofIndices(1, 2, 52, 62).toLong()));
     }
 
     // [itest->dsn~assign-roles-to-user-creates-a-role~1]
     @Test
     void testAssignRolesToUserUpdatesUserRoles() throws SQLException {
-        script.execute("NORBERT", List.of("Sales", "Development"));
-        script.execute("NORBERT", List.of("Sales"));
+        script.execute("NORBERT", List.of("role_1", "role_2"));
+        script.execute("NORBERT", List.of("role_1"));
         assertThat(query("SELECT EXA_USER_NAME, EXA_ROLE_MASK FROM " + getUserTableName()), table() //
                 .row("NORBERT", 1) //
                 .matches(NO_JAVA_TYPE_CHECK));
     }
 
     // [itest->dsn~assign-roles-to-user-creates-a-role~1]
     @ParameterizedTest
-    @MethodSource("provideValuesForTestAssignUnknownRoleToUserThrowsRoleNotFoundException")
-    void testAssignUnknownRoleToUserThrowsRoleNotFoundException(final List<String> allRoles,
-            final List<String> unknownRoles) {
-        assertScriptThrows("The following roles were not found: " + String.join(", ", unknownRoles), "THE_USER",
+    @MethodSource("provideValuesForTestAssignIllegalRoleToUserThrowsException")
+    void testAssignIllegalRoleToUserThrowsException(final List<String> allRoles, final List<String> unknownRoles) {
+        assertScriptThrows("The following role names are not valid identifiers: \""
+                + String.join("\", \"", unknownRoles) + "\". Use numbers, letters and underscores only.", "THE_USER",
                 allRoles);
     }
 
-    private static Stream<Arguments> provideValuesForTestAssignUnknownRoleToUserThrowsRoleNotFoundException() {
-        return Stream.of(Arguments.of(List.of("Cats"), List.of("Cats")), //
-                Arguments.of(List.of("Cats", "Sales"), List.of("Cats")), //
-                Arguments.of(List.of("Sales", "Cats", "Dogs"), List.of("Dogs", "Cats")));
+    private static Stream<Arguments> provideValuesForTestAssignIllegalRoleToUserThrowsException() {
+        return Stream.of(Arguments.of(List.of("/Cats"), List.of("/Cats")), //
+                Arguments.of(List.of("Cats%", "role_1"), List.of("Cats%")), //
+                Arguments.of(List.of("role_1", "Cat&&s", "Dog§§s", "Mi ce"), List.of("Cat&&s", "Dog§§s", "Mi ce")));
+    }
+
+    @ValueSource(strings = { "Rabb!it", "K@ng@roo", "El ephant" })
+    @ParameterizedTest
+    void testAssingingToIllegalUserThrowsException(final String userName) {
+        assertScriptThrows(
+                "The user name \"" + userName
+                        + "\" is not a valid identifier. Use numbers, letters and underscores only.",
+                userName, List.of("role_1"));
     }
```

Comment:
We should add a test assigning nonexistent roles to a user, especially since this behavior changed.